klippa/src/inc_bimap.rs

```diff
@@ -0,0 +1,48 @@
+//! inc_bimap: incremental bijective map, only lhs is given, rhs is incrementally assigned
+//! ported from Harfbuzz hb_inc_bimap_t: <https://github.com/harfbuzz/harfbuzz/blob/b5a65e0f20c30a7f13b2f6619479a6d666e603e0/src/hb-bimap.hh#L97>
+
+use core::slice::Iter;
+use fnv::FnvHashMap;
+
+#[derive(Default)]
+pub(crate) struct IncBiMap {
+    forw_map: FnvHashMap<u32, u32>,
+    back_map: Vec<u32>,
+}
+
+impl IncBiMap {
+    pub(crate) fn new(capacity: usize) -> Self {
```

Comment:
for symmetry with other basic data structures I would expect `new` to not take any arguments, and for there to be a `with_capacity` constructor if you wanted to pass a capacity.